### ISSUE TITLE
Edit-Font Stretch param gives (nonfatal) tcl error

### ISSUE THREAD

**Issue by chainjazz** [pure-data/pure-data#1133]
(original report redundant, see below)

**Comment by umlaeute** [comment]
1. could you please post detailed instructions on how to reproduce this error? ("(original report redundant, see below)" is **not** very helpful; i don't know how it is redundant, nor what you are referring to with "below")

> Does Pd explicitly runs its own wish/tclsh?

yes (unless i'm mistaken)

**Comment by chainjazz** [comment]
1., set logging to level 4 all
2. Open a patch window.
3. Select Edit -> Font menu item
4.. Change the "Stretch" field to something else then 100 (in my case 125)
5. Click apply or OK
6. Error

The "below" refers to the next comment in order as it appears on GitHub website by default.

The redundancy part, I had copied the complete log which featured the same error message repeated over and over, before updating to the latest version, so in order to make it easier to read the comment feed, I put I note (also, couldn't find how to remove the initial comment)

**Comment by umlaeute** [comment]
> 5. Error

not here.
i'm assuming that with "Open a patch window" you really mean <kbd>Control</kbd>-<kbd>n</kbd> (to get a new, empty patch window).
if not, please provide a minimal patch that triggers the problem.

> 6.1 If you don't see the error message, try changing the font via command line parameters -font-face, etc.

no, seriously.
please provide a full example that triggers the problem.
if you use custom startup flags, you must provide **all** of them, so we can reproduce the problem.
if you use a special font, you must say so.

make sure you run Pd with `-noprefs`.
uninstall/disable all GUI-plugins that you happen to have.
don't load any externals.


> The "below" refers to the next comment in order as it appears on GitHub website by default.
[...]
> (also, couldn't find how to remove the initial comment)

so why don't you just put the updated information in place of your original comment?

**Comment by sebshader** [comment]
it happens when there's a GOP object in a (non-abstraction) subpatch

**Comment by sebshader** [comment]
it happens when there's a gop object connected to another object in a non-abstraction subpatch of the patch whose "font" dialog it is

**Comment by umlaeute** [comment]
it would be fantastic if you could post such a patch here, with in-patch instructions (with numbers, if the order is important) on where you need to do what.

**Comment by umlaeute** [comment]
like so:

~~~
#N canvas 749 176 450 300 12;
#N canvas 915 331 450 300 fuu 0;
#N canvas 5 53 450 300 bar 0;
#X obj 289 130 inlet;
#X obj 289 162 outlet;
#X connect 0 0 1 0;
#X coords 0 -1 1 1 85 60 1 100 100;
#X restore 128 132 pd bar;
#X obj 128 216 print;
#X connect 0 0 1 0;
#X restore 69 60 pd fuu;
#X text 50 118 1 open the "Font" dialog for this window;
#X text 50 143 2 change the "stretch" value to 120;
#X text 50 170 3 click on "OK";
~~~

**Comment by sebshader** [comment]
#N canvas 332 121 450 300 10;
#N canvas 238 163 450 300 test 0;
#N canvas 0 23 450 300 agraph 0;
#X obj 215 195 outlet;
#X coords 0 -1 1 1 85 60 1 100 100;
#X restore 122 31 pd agraph;
#X obj 133 126 bang;
#X connect 0 0 1 0;
#X restore 83 201 pd test;
#X text 32 42 1 select edit->font;
#X text 43 79 2 edit stretch field;
#X text 60 109 3 click "ok";
#X text 140 199 <- simple gop connected inside;

**Comment by sebshader** [comment]
somewhere in canvas_dofont within g_editor.c I guess. It doesn't seem to happen with GOP (non-abstraction) subpatches

**Comment by sebshader** [comment]
yes, it happens with gatoms to when they try to move their labels. It doesn't happen for g_texts and iemgui objects since they check if the glist is visible before sending drawing instructions

**Comment by umlaeute** [comment]
included in `master` with 2b5c219656771a311f36d8ac826281f153e0ea3e
